src/stiefel.jl

```diff
@@ -169,19 +169,17 @@ function stiefelexp(W::AbstractTensorMap,
                     α::Real)
     S = spacetype(W)
     G = sectortype(W)
-    Wdata′ = TensorKit.SectorDict{G,storagetype(W)}()
-    Qdata = TensorKit.SectorDict{G,storagetype(W)}()
-    Qdata′ = TensorKit.SectorDict{G,storagetype(W)}()
-    Rdata′ = TensorKit.SectorDict{G,storagetype(W)}()
     dims = TensorKit.SectorDict{G,Int}()
-    for c in blocksectors(W)
-        w′, q, q′, r′ = _stiefelexp(block(W, c), block(A, c), block(Z, c), α)
-        Wdata′[c] = w′
-        Qdata[c] = q
-        Qdata′[c] = q′
-        Rdata′[c] = r′
+    generator = Iterators.map(blocks(W)) do (c, b)
+        w′, q, q′, r′ = _stiefelexp(b, block(A, c), block(Z, c), α)
         dims[c] = size(q, 2)
+        return c => (w′, q, q′, r′)
     end
+    wqqr = TensorKit.SectorDict(generator)
+    Wdata′ = SectorDict(c => w′ for (c, (w′, q, q′, r′)) in wqqr)
+    Qdata = SectorDict(c => q for (c, (w′, q, q′, r′)) in wqqr)
+    Qdata′ = SectorDict(c => q′ for (c, (w′, q, q′, r′)) in wqqr)
+    Rdata′ = SectorDict(c => r′ for (c, (w′, q, q′, r′)) in wqqr)
```

Comment:
This seems very similar to what I do for the `tsvd` in TensorKit.jl, but there I need to be able to take truncation into account. I think for this, I would prefer a paradigm as follows:
```julia
dims = SectorDict(c=>size(b, 2) for (c,b) in blocks(W))
V = S(dims)
W′ = similar(W)
Q = similar(W, codomain(W) ← V)
Q′ = similar(Q)
R′ = similar(W, V ← domain(W))
for (c, b) in blocks(W)
   w′, q, q′, r′ = _stiefelexp(b, block(A, c), block(Z, c), α)
   copy!(block(W′, c), w′)
   copy!(block(Q, c), q)
   copy!(block(Q′, c), q′)
   copy!(block(R′, c), r′)
end
```
This makes it easier to at some later point transform this into a `_stiefelexp!` method which immediately stores the output in the right memory location and thus doesn't require the explicit `copy!` operations.

I now also noticed that this is closer in style to the next change below as well.

Reply:
I'll change it. I didn't know that `V` can just be found from the inputs, which is why I mimicked the truncated svd code, but I also prefer this approach. (would this also work: `V = fuse(domain(W))`?)